homeassistant/components/noaa_tides/sensor.py

```diff
@@ -51,24 +51,31 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     else:
         unit_system = UNIT_SYSTEMS[0]
 
-    noaa_sensor = NOAATidesAndCurrentsSensor(name, station_id, timezone, unit_system)
+    station = coops.Station(station_id, unit_system)
 
-    noaa_sensor.update()
-    if noaa_sensor.data is None:
-        _LOGGER.error("Unable to setup NOAA Tides Sensor")
+    if station.name is None:
+        _LOGGER.error(
+            "Unable to setup NOAA Tides Sensor for station_id: %s", station_id
+        )
```

Comment:
You should raise `PlatformNotReady` if you are unable to setup.  If there was a momentary issue on connectivity, this will ensure that home assistant retries setup later.

Reply:
Thanks for the tip. I think it is a better idea to catch exceptions in the line above and just get rid of this altogether. I'll raise `PlatformNotReady` there where appropriate.